mongoctl/commands/common/connect.py

```diff
@@ -97,9 +98,12 @@ def open_mongo_shell_to_server(server,
         else:
             database = "admin"
 
-    if username or server.needs_to_auth(database):
+    login_database = database if not "authenticationDatabase" in shell_options \
```

Comment:
Where this idiom is used elsewhere, the default value comes after the `else`, so I think reversing what you have here is more consistent:
```
login_database = shell_options["authenticationDatabase"] if "authenticationDatabase" in shell_options else database
```